jsx/StaticDataTable.js

```diff
@@ -303,6 +303,54 @@ class StaticDataTable extends Component {
   getSortedRows() {
     const index = [];
 
+    // TODO: data column type check should probably be done once at init,
+    // to find which columns have mixed types.
+    // First pass to check if there is a mix of types in the values
+    const typesFound = [];
```

Comment:
If you don't need the list of types found at the end, what I would do is have a `null` variable, assign the found type on the first iteration, and check each other iteration on that type (instead of having an array lookup on each iteration). But your approach can also work.

Reply:
and if you do need the types at the end using an object instead of an array would be a constant time lookup for each iteration instead of linear.

Reply:
Agreed. But if we are to really nitpick, I think the [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) object would be even more appropriate in this situation. The constant-time vs linear-time lookup is an interesting theoretic point, but would likely not hold here because of the small size of the collection (I would even bet the array is faster here).

Please note that this is a purely hypothetical discussion and I still recommend the use of a simple variable in this PR.

Reply:
Seeing it now, a single var will be enough.